tests/test_tokenizers/test_stanza_tokenizer.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+from rasa.nlu.training_data import Message, TrainingData
+from rasa.nlu.constants import TEXT, TOKENS_NAMES, SPARSE_FEATURE_NAMES
+from rasa.nlu.tokenizers.whitespace_tokenizer import WhitespaceTokenizer
+from rasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer import (
+    CountVectorsFeaturizer,
+)
+from rasa.nlu.featurizers.sparse_featurizer.lexical_syntactic_featurizer import (
+    LexicalSyntacticFeaturizer,
+)
+from rasa_nlu_examples.tokenizers.stanzatokenizer import StanzaTokenizer
+
+
+@pytest.mark.parametrize(
+    "msg,n", zip(["hello", "hello there", "hello there vincent"], [1, 2, 3])
+)
+def test_stanza_correct_length(msg, n):
+    """We should add the correct number of tokens."""
+    message = Message(msg)
+    tok = StanzaTokenizer(
+        component_config={"lang": "en", "cache_dir": "tests/data/stanza"}
+    )
+    tok.process(message)
+    tokens = message.get(TOKENS_NAMES[TEXT])
+    # We also generate a __CLS__ token
+    assert len(tokens) == n + 1
+
+
+def test_stanza_lemma():
+    """We need to attach correct lemmas"""
+    message = Message("i am running and giving many greetings")
+    tok = StanzaTokenizer(
+        component_config={"lang": "en", "cache_dir": "tests/data/stanza"}
+    )
+    tok.process(message)
+    tokens = message.get(TOKENS_NAMES[TEXT])
+    assert [t.lemma for t in tokens] == [
+        "i",
+        "be",
+        "run",
+        "and",
+        "give",
+        "many",
+        "greeting",
+        "__CLS__",
+    ]
+
+
+def test_stanza_pos():
+    """We need to attach correct POS"""
+    message = Message("i am running and giving many greetings")
+    tok = StanzaTokenizer(
+        component_config={"lang": "en", "cache_dir": "tests/data/stanza"}
+    )
+    tok.process(message)
+    tokens = message.get(TOKENS_NAMES[TEXT])
+    assert [t.data.get("pos", "") for t in tokens] == [
+        "PRON",
+        "AUX",
+        "VERB",
+        "CCONJ",
+        "VERB",
+        "ADJ",
+        "NOUN",
+        "",
+    ]
+
+
+def fetch_sparse_features(txt, tokenizer, featurizer):
```

Comment:
What are you testing here from the tokenizer point of view?

Reply:
This is not a test, this is a function used in the two tests below.

Reply:
ups, sorry 🤦